subprojects/tooling-api/src/main/java/org/gradle/tooling/exceptions/MultipleBuildActionsException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.exceptions;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Thrown when the {@link org.gradle.tooling.PhasedBuildActionExecuter.Builder} has been configured
+ * with more than one {@link org.gradle.tooling.BuildAction} per phase.
+ *
+ * @since 4.7
+ */
+@Incubating
+public class MultipleBuildActionsException extends RuntimeException {
```

Comment:
❌ This is overly specific in my opinion. This is a programming error, not something that should be specially reacted to. A simple IllegalArgumentException should do.